src/app/app.component.ts

```diff
@@ -23,3 +27,87 @@ export class AppComponent {
 
   }
 }
+
+@Component({
+  selector: 'app-calendar-header',
+  template: `
+    <mat-form-field>
+      <mat-select placeholder="Calendar Format" [formControl]="form.controls['calendar']">
+        <mat-option *ngFor="let cal of supportedCalendarFormats" [value]="cal">{{cal}}</mat-option>
+      </mat-select>
+    </mat-form-field>
+    <div class="custom-header">
```

Comment:
@mmalerba If I try to use `<mat-calendar-header></mat-calendar-header>` instead I get

```
compiler.js:215 Uncaught Error: Template parse errors:
No provider for MatCalendar ("
      <button mat-icon-button (click)="nextClicked('year')">&gt;&gt;</button>
    </div>
    [ERROR ->]<mat-calendar-header></mat-calendar-header>
  "): ng:///AppModule/HeaderComponent.html@13:4
    at syntaxError (compiler.js:215)
    at TemplateParser.parse (compiler.js:14692)
    at JitCompiler._parseTemplate (compiler.js:22632)
    at JitCompiler._compileTemplate (compiler.js:22619)
    at eval (compiler.js:22562)
    at Set.forEach (<anonymous>)
    at JitCompiler._compileComponents (compiler.js:22562)
    at eval (compiler.js:22472)
    at Object.then (compiler.js:206)
    at JitCompiler._compileModuleAndComponents (compiler.js:22471)
```

However, if I use `<mat-year-view></mat-year-view>`it works just fine.

I am using `6.0.0-rc.3`.

Is there still something missing to export `MatCalendarHeader`? Or is the export you made  not included yet in `6.0.0-rc.3`?

Reply:
WebStorm complains about `mat-calendar-header`:

<img width="1037" alt="mat-calendar-header" src="https://user-images.githubusercontent.com/6000023/38742645-6f991706-3f3d-11e8-95f0-97a7caa124aa.png">

Reply:
```
/** Default header for MatCalendar */
@Component({
  moduleId: module.id,
  selector: 'mat-calendar-header',
  templateUrl: 'calendar-header.html',
  encapsulation: ViewEncapsulation.None,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
```

```
/**
 * An internal component used to display a single year in the datepicker.
 * @docs-private
 */
@Component({
  moduleId: module.id,
  selector: 'mat-year-view',
  templateUrl: 'year-view.html',
  exportAs: 'matYearView',
  encapsulation: ViewEncapsulation.None,
  changeDetection: ChangeDetectionStrategy.OnPush
})
```

There is no  `exportAs` for `MatCalendarHeader`. Is that needed to use the selector outside of material?